DIOSSession.m

```diff
@@ -67,13 +75,20 @@ - (NSMutableURLRequest *) requestWithMethod:(NSString *)method
                                  parameters:(NSDictionary *)parameters
 {
     NSString *urlString = [NSString stringWithFormat:@"%@/%@", [[self baseURL] absoluteString], path];
+    NSLog(@"Urlstring is %@", urlString);
     [self.requestSerializer setValue:@"application/json" forHTTPHeaderField:@"Accept"];
     [self.requestSerializer setValue:@"application/hal+json" forHTTPHeaderField:@"Content-Type"];
 
     if (self.signRequests) {
-        //doesntwork
+        
         [self.requestSerializer setAuthorizationHeaderFieldWithUsername:_basicAuthUsername password:_basicAuthPassword];
     }
+    else{
+        // If previously signRequest is set to YES then if temparoraly it has been set to NO then authorization header should be cleared
```

Comment:
Can you rephrase this sentence or make it more complete? Im having a hard time reading it.